pkg/distributor/ratestore_test.go

```diff
@@ -84,32 +78,26 @@ func TestRateStore(t *testing.T) {
 
 		tc.clientPool.clients = map[string]client.PoolClient{
 			"ingester0": newRateClient([]*logproto.StreamRate{
-				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 25},
-				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 25},
+				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 25, Pushes: 35},
+				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 25, Pushes: 35},
 			}),
 			"ingester1": newRateClient([]*logproto.StreamRate{
-				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 35},
-				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 35},
+				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 35, Pushes: 10},
+				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 35, Pushes: 10},
 			}),
 			"ingester2": newRateClient([]*logproto.StreamRate{
-				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 15},
-				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 15},
+				{Tenant: "tenant 1", StreamHash: 0, StreamHashNoShard: 0, Rate: 15, Pushes: 25},
+				{Tenant: "tenant 2", StreamHash: 0, StreamHashNoShard: 0, Rate: 15, Pushes: 25},
 			}),
 		}
 
-		_ = tc.rateStore.StartAsync(context.Background())
-		defer tc.rateStore.StopAsync()
-
-		require.Eventually(t, func() bool { // There will be data
-			return tc.rateStore.RateFor("tenant 1", 0) != 0 &&
-				tc.rateStore.RateFor("tenant 2", 0) != 0
-		}, time.Second, time.Millisecond)
+		require.NoError(t, tc.rateStore.instrumentedUpdateAllRates(context.Background()))
 
-		require.Equal(t, int64(35), tc.rateStore.RateFor("tenant 1", 0))
-		require.Equal(t, int64(35), tc.rateStore.RateFor("tenant 2", 0))
+		requireRatesAndPushesEqual(t, 35, 10, tc.rateStore, "tenant 1", 0)
+		requireRatesAndPushesEqual(t, 35, 10, tc.rateStore, "tenant 1", 0)
```

Comment:
tenant 2 also expects 35 as the highest rate
```suggestion
		requireRatesAndPushesEqual(t, 35, 10, tc.rateStore, "tenant 2", 0)
```